src/zero/module/aou.clj

```diff
@@ -1,23 +1,400 @@
+   :gender_cluster_file         "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExomev1_1_gender.egt"
+   :zcall_thresholds_file       "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/IBDPRISM_EX.egt.thresholds.txt"}
+  )
+
+(def fingerprinting
+  "Fingerprinting inputs for arrays."
+  {:fingerprint_genotypes_vcf_file       nil
+   :fingerprint_genotypes_vcf_index_file nil
+   :haplotype_database_file              "gs://gcp-public-data--broad-references/hg19/v0/Homo_sapiens_assembly19.haplotype_database.txt"
+   :variant_rsids_file                   "gs://broad-references-private/hg19/v0/Homo_sapiens_assembly19.haplotype_database.snps.list"}
+  )
+
+(def genotype-concordance
+  "Genotype Concordance inputs for arrays."
+  {:control_sample_vcf_file       "gs://broad-gotc-test-storage/arrays/controldata/NA12878.vcf.gz"
+   :control_sample_vcf_index_file "gs://broad-gotc-test-storage/arrays/controldata/NA12878.vcf.gz.tbi"
+   :control_sample_intervals_file "gs://broad-gotc-test-storage/arrays/controldata/NA12878.interval_list"
+   :control_sample_name           "NA12878"})
+
+(def other-inputs
+  "Miscellaneous inputs for arrays."
+  {:call_rate_threshold              0.98
+   :genotype_concordance_threshold   0.98
+   :contamination_controls_vcf       nil
+   :subsampled_metrics_interval_list nil
+   :disk_size                        100
+   :preemptible_tries                3})
+
+(defn- map-aou-environment
+  "Map AOU-ENV to envrionment for inputs preparation."
+  [aou-env]
+  ({:aou-dev "dev" :aou-prod "prod"} aou-env))
+
+(defn array-inputs
+  "Array inputs for ENVIRONMENT that do not depend on the input file."
+  [environment]
+  {:vault_token_path (get-in env/stuff [environment :vault_token_path])
+   :environment      (map-aou-environment environment)})
+
+(comment
+  (array-inputs :aou-dev))
+
+(defn extract-and-validate-per-sample-inputs
+  "Extract valid per-sample inputs from OBJECT"
+  [object]
+  (let [per-sample (select-keys object [:analysis_version_number
+                                        :bead_pool_manifest_file
+                                        :chip_well_barcode
+                                        :cluster_file
+                                        :extended_chip_manifest_file
+                                        :gender_cluster_file
+                                        :green_idat_cloud_path
+                                        :params_file
+                                        :red_idat_cloud_path
+                                        :reported_gender
+                                        :sample_alias
+                                        :sample_lsid
+                                        :zcall_thresholds_file])]
+    (if (every? (fn [x] (not (nil? x))) (vals per-sample))
+      per-sample
+      (throw (Exception. (format "Invalid sample inputs %s" per-sample))))))
```

Comment:
Suggest: https://github.com/broadinstitute/wfl/blob/tbl/rex/GH-671-Build-Arrays-Module/src/zero/module/aou.clj#L95-L114

Reply:
+1 to return filtered inputs and throw missing.
btw, the link doesn't run: 
```
Execution error (ArityException) at zero.module.aou/eval10123 (aou.clj:102).
Wrong number of args (3) passed to: clojure.lang.PersistentVector
```

Reply:
Sorry. That's what I get for not running it. It should work better now.

Reply:
Thanks! It works now with 1 tiny hack:
```diff
(defn get-per-sample-inputs
  "Throw or return per-sample INPUTS."
  [inputs]
  (let [the-keys [:analysis_version_number
                  :bead_pool_manifest_file
                  :chip_well_barcode
                  :cluster_file
                  :extended_chip_manifest_file
                  :gender_cluster_file
                  :green_idat_cloud_path
                  :params_file
                  :red_idat_cloud_path
                  :reported_gender
                  :sample_alias
                  :sample_lsid
                  :zcall_thresholds_file]
        result  (select-keys inputs the-keys)
-        missing (vec (keep (fn [k] (when (nil? (k result))) k) the-keys))]
+        missing (vec (keep (fn [k] (when (nil? (k result))k)) the-keys))]
    (when (seq missing)
      (throw (Exception. (format "Missing per-sample inputs: %s" missing))))
    result))
```